brush-parser/src/parser.rs

```diff
@@ -847,27 +847,31 @@ fn add_pipe_extension_redirection(c: &mut ast::Command) -> Result<(), &'static s
         ast::IoFileRedirectKind::DuplicateOutput,
         ast::IoFileRedirectTarget::Fd(1),
     );
+
+    #[inline]
```

Comment:
Why `#[inline]`? For what it's worth, the best write-up I've seen with guidance on this is here: https://matklad.github.io/2021/07/09/inline-in-rust.html.

Reply:
This function is for not duplicating both match branches, but I would prefer not to write a `macro!` for this. So I decided to tell the compiler to inline the function body. Actually it might inline it anyway... so I can remove it...

Reply:
>Actually it might inline it anyway... so I can remove it...

Okay, please remove it and then I can do a final review.

Taking inspiration from the page linked above, it makes sense to avoid proactively tagging functions for inlining unless we analyze performance and identify that it would make a meaningful difference. (For what it's worth, release builds of `brush` do have link-time optimization turned on as well.)